syntax = "proto3";


message Product {
    int32 product_id = 1;
    string product_name =2;            
    string description = 3;      
    float price = 4;             
    string expiry = 5;           
    string brand = 6;
    string category = 7;         
}



// Enum for different operation types
enum OperationType {
    CREATE = 0;   
    UPDATE = 1;   
    DELETE = 2;   
}

// Message structure for product operations
message ProductMessage {
    OperationType message_type = 1; 
    Product product_data = 2;        
    int32 product_id = 3;            
}
